registry/memory/memory.go

```diff
@@ -74,91 +118,152 @@ func (m *Registry) Init(opts ...registry.Option) error {
 
 	// add services
 	m.Lock()
-	for k, v := range getServices(m.options.Context) {
-		s := m.Services[k]
-		m.Services[k] = registry.Merge(s, v)
+	defer m.Unlock()
+
+	records := getServiceRecords(m.options.Context)
+	for name, record := range records {
+		// add a whole new service including all of its versions
+		if _, ok := m.records[name]; !ok {
+			m.records[name] = record
+			continue
+		}
+		// add the versions of the service we dont track yet
+		for version, r := range record {
+			if _, ok := m.records[name][version]; !ok {
+				m.records[name][version] = r
+				continue
+			}
+		}
 	}
-	m.Unlock()
+
 	return nil
 }
 
 func (m *Registry) Options() registry.Options {
 	return m.options
 }
 
-func (m *Registry) GetService(name string) ([]*registry.Service, error) {
-	m.RLock()
-	service, ok := m.Services[name]
-	m.RUnlock()
-	if !ok {
-		return nil, registry.ErrNotFound
+func (m *Registry) Register(s *registry.Service, opts ...registry.RegisterOption) error {
+	m.Lock()
+	defer m.Unlock()
+
+	var options registry.RegisterOptions
+	for _, o := range opts {
+		o(&options)
 	}
 
-	return service, nil
-}
+	// if no TTL has been set, set it to DefaultTTL
+	if options.TTL.Seconds() == 0.0 {
+		options.TTL = DefaultTTL
```

Comment:
If a TTL is not set we don't set it here because there's no guarantee the caller will call Register again to update the TTL.

Reply:
what is the DefaultTTL then? If the client does not set it (unless it sets it be default) then TTL is set to 0 and it will be pruned by `m.registry.ttlPrune()`

Reply:
There is no default TTL. Records can be stored indefinitely. This is the behaviour of all other registries.

Reply:
If the TTL is not set you'd not try to prune it